src/matching_market.sol

```diff
@@ -489,7 +487,7 @@ contract MatchingMarket is MatchingEvents, ExpiringMarket, DSNote {
         address pay_gem = address(offers[id].pay_gem);
         require(_span[pay_gem][buy_gem] > 0);
 
-        require(_rank[id].delb == 0 && isOfferSorted(id));  // Assert id is in the sorted list
+        require(isActive(id));                              // Assert id is active, then it is in the sorted list
```

Comment:
Why we dont check for `rank[id].delb`?

Reply:
If the offer is active, then `delb == 0` and is sorted always.